src/core/scenario.jl

```diff
@@ -111,22 +111,77 @@ function Scenario(d::Dict; flex_hvac_from_json=false)
+                                            latitude=site.latitude, longitude=site.longitude, 
+                                            time_steps_per_hour=settings.time_steps_per_hour
+                                        )
     elseif !(settings.off_grid_flag)
-        electric_utility = ElectricUtility()
+        electric_utility = ElectricUtility(; latitude=site.latitude, longitude=site.longitude, 
+                                            time_steps_per_hour=settings.time_steps_per_hour
+                                        )
     elseif settings.off_grid_flag 
         if haskey(d, "ElectricUtility")
             @warn "ElectricUtility inputs are not applicable when off_grid_flag is true and any ElectricUtility inputs will be ignored. For off-grid scenarios, a year-long outage will always be modeled."
         end
-        electric_utility = ElectricUtility(; outage_start_time_step = 1, outage_end_time_step = settings.time_steps_per_hour * 8760) 
+        electric_utility = ElectricUtility(; outage_start_time_step = 1, 
+                                            outage_end_time_step = settings.time_steps_per_hour * 8760, 
+                                            latitude=site.latitude, longitude=site.longitude, 
+                                            time_steps_per_hour=settings.time_steps_per_hour
+                                        ) 
     end
 
+    #Handle missing emissions inputs (due to failed lookup and not provided by user)
+    #TODO: condition below code on: if site.off_grid == false
+    if isnothing(electric_utility.emissions_factor_series_lb_CO2_per_kwh)
+        if (!isnothing(site.CO2_emissions_reduction_min_pct) || 
+            !isnothing(site.CO2_emissions_reduction_max_pct) || 
+            settings.include_climate_in_objective)
+            throw(@error "To include CO2 costs in the objective function or enforce emissions reduction constraints, you must either enter custom health emissions factors or a site location within the continental U.S.")
+        else
+            electric_utility.emissions_factor_series_lb_CO2_per_kwh = zeros(8760*settings.time_steps_per_hour)
+        end
+    end
+    for emissions_type in ["NOx", "SO2", "PM25"]
+        if isnothing(getproperty(electric_utility, Symbol("emissions_factor_series_lb_$(emissions_type)_per_kwh")))
+            if settings.include_health_in_objective
+                throw(@error "To include health costs in the objective function, you must either enter custom health emissions factors or a site location within the continental U.S.")
+            else
+                setproperty!(electric_utility, 
+                    Symbol("emissions_factor_series_lb_$(emissions_type)_per_kwh"), 
+                    zeros(8760*settings.time_steps_per_hour)
+                )
+            end
+        end
+    end
+    #TODO: allow grid costs to be missing if site.off_grid == true
```

Comment:
Can we do this now?

Reply:
@hdunham @adfarth should EASIUR based grid_costs, onsite_costs and escalation_rates only be set if off_grid_flag is false?

Reply:
done in 7dd4fe7

Reply:
in 9b613d4, not looking up grid costs if off grid